pkgs/applications/virtualization/rust-hypervisor-firmware/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, fetchFromGitHub
+, makeRustPlatform
+, hostPlatform
+, targetPlatform
+, lld
+}:
+
+let
+  arch = targetPlatform.qemuArch;
+
+  target = ./. + "/${arch}-unknown-none.json";
+
+  cross = import ../../../.. {
+    system = hostPlatform.system;
+    crossSystem = lib.systems.examples."${arch}-embedded" // {
+      rustc.config = "${arch}-unknown-none";
+      rustc.platform = lib.importJSON target;
+    };
+  };
+
+  inherit (cross) rustPlatform;
+
+in
+
+rustPlatform.buildRustPackage rec {
+  pname = "rust-hypervisor-firmware";
+  version = "0.4.2";
+
+  src = fetchFromGitHub {
+    owner = "cloud-hypervisor";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-hKk5pcop8rb5Q+IVchcl+XhMc3DCBBPn5P+AkAb9XxI=";
+  };
+
+  cargoSha256 = "sha256-edi6/Md6KebKM3wHArZe1htUCg0/BqMVZKA4xEH25GI=";
+
+  RUSTC_BOOTSTRAP = 1;
```

Comment:
hm why do we need bootstrap here? is it unstable-only or because of UEFI?
if UEFI, please add a comment "remove me when 1.68.1 lands" which is anytime soon because it's in staging right now

Reply:
Without it, r-efi lacks `abi_efiapi`.

Looking forward towards Rust 1.68, I am considering to bump this package to unstable because it might build on aarch64 then.

Reply:
We can keep RUSTC_BOOTSTRAP for aarch64 and riscv64 then. :)